File: 宏图培训机构作业练习/JAVA练习/课外没事写一写身强体壮我最棒/菱形.java
public class 菱形
{
	public static void main(String dls[]){
		System.out.println("固定的行数的菱形");
		int h=17,mid=h/2+1,i=1,j=1,k=1,x=1;
		for(i=1;i<=mid;i++){
			x=2*i-1;
			k=(h-x)/2;
			for(j=1;j<=k;j++){
				System.out.print(" ");
			}
			for(j=1;j<=x;j++){
				System.out.print("*");
			}
			System.out.print("\n");
		}
		for(i=mid-1;i>0;i--){
			x=2*i-1;
			k=(h-x)/2;
			for(j=1;j<=k;j++){
				System.out.print(" ");
			}
			for(j=1;j<=x;j++){
				System.out.print("*");
			}
			System.out.print("\n");
		}
	}
}

File: 宏图培训机构作业练习/C语言练习/第六讲/老师做的标准必会8题/菱形.c
/* Note:Your choice is C IDE */
#include "stdio.h"
void main()
{
/*
		*
	   ***
	  *****
	 *******
	  *****
	   ***
	    *
	1、行号等于字符数
	2、星号的个数：行数*2-1
	3、空格个数：（字符数-星号数）/2
*/
	//输出菱形 
    int line,start,space,i,j,mid;
    printf("请输入行号：（奇数）");
    scanf("%d",&line);
    mid=line/2+1;
    for(i=1;i<=mid;i++){ //控制菱形的行数
    	//计算星号个数
    	start=i*2-1;
    	//打印空格
    	space=(line-start)/2;
    	for(j=1;j<=space;j++){
    		printf(" ");	
    	}
    	//打印星号
    	for(j=1;j<=start;j++){
    		printf("*");	
    	}
    	printf("\n");
    }
    //
    for(i=mid-1;i>0;i--){ //控制菱形的行数
    	//计算星号个数
    	start=i*2-1;
    	//打印空格
    	space=(line-start)/2;
    	for(j=1;j<=space;j++){
    		printf(" ");	
    	}
    	//打印星号
    	for(j=1;j<=start;j++){
    		printf("*");	
    	}
    	printf("\n");
    }
}